api/server.js

```diff
@@ -37,22 +38,31 @@ const esClient = new elasticsearch.Client({
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
-app.get('/api/resources/ingredients', async (_, response) => {
+app.get('/api/resources/ingredients', asyncHandler(async (request, resoponse, next) => {
   response.send({
     items: ingredients.ingredients,
   });
-});
+}));
 
+function isString (value) {
+  return typeof value === 'string' || value instanceof String;
+}
 
-app.post('/api/search/ingredients', async (request, resoponse) => {
+app.post('/api/search/recipes', asyncHandler(async (request, resoponse, next) => {
   const {
+    freeText,
     includeTerms,
     excludeTerms,
+    fromCookTime = 0,
+    toCookTime = 600,
+    tags,
```

Comment:
Where do the tags and diet come from the user input? I imagined the freeText would search the 'tags' and 'diet' fields and match to them?

Reply:
So this is an open question. In our mock ups we have selection fields for meal type and dietary prefs, so I thought it would come from them.
Now free text should search everything of content in my opinion, I've already put in title and ingredients, will add now tags and diet, I just don't have them in my Elastic instance.

Reply:
@LeshemNoa are we still planing on having those selection fields in the sidebar?

Reply:
I imagine it'll be fine to keep them like this as long as we give the user a list of dietary preferences and meal types to choose from (Which is what most websites do)

Also I think that the freeText section should also match/search from the tags section, not just the ingredients/title

Reply:
free text should search the recipe title, e.g. when you want to make soup or salad or whatever. 
All other fields should be in the sidebar as planned

Reply:
The question is which fields does the string search work on. 
The search bar should enable you to search for any search terms you want, whether it is some ingredients, type of meal and diet. Limiting it to only the title makes it way less effective. If I want to search for example a "Vegan Sandwich" I want results which might not have vegan in the title but since they are vegan, have it in their tags. That's why I think we should work on title, ingredients, tags and diet.

Reply:
I agree with Hagai here, I think we lose a lot of potential value by not allowing the free text to interact with the the tags - There can be stuff in there outside of seasonal/dietary elements that can help the user get to what he really wants.

On the same note, do you feel the need for a user to be able to search by website? IMO we shouldn't do that because that would make our site somewhat obsolete, because if a user only wants recipes from a specific site, he could just go to that site.

Also on the frontend - although this pertains more to the cards themselves - one of our next upgrades should be to make the cards more detailed in terms of understanding what the recipe is - main ingredients, what website it's from, cooking time and rating if possible

Reply:
limiting your search to several providers is certainly a feature but we need to draw the line somewhere and I think it's here.
Of course that info has to be presented like we were asked to do inn the last milestone feedback.